ecl/hql/hqlmanifest.cpp

```diff
@@ -198,6 +198,31 @@ void ResourceManifest::loadResource(const char *filepath, MemoryBuffer &content)
     read(fio, 0, (size32_t) f->size(), content);
 }
 
+void queryJfrog(const char *localpath, IPropertyTree &item)
+{
+    const char *repo = item.queryProp("@repository");
+    if(repo)
+    {
+        StringBuffer filename(localpath);
+        getFileNameOnly(filename, false);
+        StringBuffer filepath;
+        StringBuffer machine;
+        if (repo[0] == '/' && repo[1] == '/')
+            splitUNCFilename(repo, &machine, &filepath, nullptr, nullptr);
+        else 
+            splitFilename(repo, nullptr, &filepath, nullptr, nullptr);
+        StringBuffer jfrog_cmd("export CI=true; jf rt dl --flat --quiet ");
+        if (!machine.isEmpty())
+            jfrog_cmd.appendf("--url=http:%s/artifactory ", machine.str());
+        StringBuffer secretValue;
+        const char *user = item.queryProp("@user");
+        if (user && getSecretValue(secretValue.clear(), "jfrog", user, "key", true))
+            jfrog_cmd.appendf("--user=%s --password=%s ", user, secretValue.trimRight().str());
+        jfrog_cmd.appendf("\"%s%s\" %s >/dev/null 2>&1", filepath.str(), filename.str(), localpath);
+        system(jfrog_cmd.str());
+    }
```

Comment:
We also need an option to verify that the hash for an entry we have downloaded matches a hash specified in the resource.

Reply:
Added option for adding md5 hash of resource in callJfrog.